src/responder/ifp/ifp_users.c

```diff
@@ -461,6 +461,304 @@ static int ifp_users_list_copy(struct ifp_list_ctx *list_ctx,
+                                                              result->msgs[0]);
+    if (name_and_cert_ctx->user_opath == NULL) {
+        error = sbus_error_new(sbus_req, SBUS_ERROR_INTERNAL,
+                               "Failed to compose object path");
+        goto fail;
+    }
+
+    if (name_and_cert_ctx->list_ctx != NULL) {
+        ret = ifp_users_find_by_name_and_cert_step(name_and_cert_ctx);
+        if (ret != EOK) {
+            error = sbus_error_new(sbus_req, DBUS_ERROR_FAILED,
+                                   "Failed to fetch certificate [%d]: %s\n",
+                                   ret, sss_strerror(ret));
+            goto fail;
+        }
+    } else {
+        ifp_users_find_by_name_and_cert_reply(name_and_cert_ctx);
+    }
+
+    return;
+
+fail:
+    talloc_free(name_and_cert_ctx);
+    sbus_request_fail_and_finish(sbus_req, error);
+    return;
+}
+
+static int ifp_users_find_by_name_and_cert_step(
+                                    struct name_and_cert_ctx *name_and_cert_ctx)
+{
+    struct tevent_req *req;
+    struct ifp_list_ctx *list_ctx = name_and_cert_ctx->list_ctx;
+
+    req = cache_req_user_by_cert_send(list_ctx,
+                                      list_ctx->ctx->rctx->ev,
+                                      list_ctx->ctx->rctx,
+                                      list_ctx->ctx->rctx->ncache,
+                                      0,
+                                      list_ctx->dom->name,
+                                      list_ctx->filter);
+    if (req == NULL) {
+        return ENOMEM;
+    }
+
+    tevent_req_set_callback(req, ifp_users_find_by_name_and_cert_done,
+                            name_and_cert_ctx);
+
+    return EOK;
+}
+
+static void ifp_users_find_by_name_and_cert_done(struct tevent_req *req)
+{
+    DBusError *error;
+    struct name_and_cert_ctx *name_and_cert_ctx;
+    struct ifp_list_ctx *list_ctx;
+    struct sbus_request *sbus_req;
+    struct cache_req_result *result;
+    errno_t ret;
+
+    name_and_cert_ctx = tevent_req_callback_data(req, struct name_and_cert_ctx);
+    list_ctx = name_and_cert_ctx->list_ctx;
+    sbus_req = list_ctx->sbus_req;
+
+    ret = cache_req_user_by_cert_recv(name_and_cert_ctx, req, &result);
+    talloc_zfree(req);
+    if (ret != EOK && ret != ENOENT) {
+        error = sbus_error_new(sbus_req, DBUS_ERROR_FAILED, "Failed to fetch "
+                               "user [%d]: %s\n", ret, sss_strerror(ret));
+        sbus_request_fail_and_finish(sbus_req, error);
+        return;
+    }
+
+    if (ret == EOK) {
+        ret = ifp_users_list_copy(list_ctx, result->ldb_result);
+        if (ret != EOK) {
+            error = sbus_error_new(sbus_req, SBUS_ERROR_INTERNAL,
+                                   "Failed to copy domain result");
+            sbus_request_fail_and_finish(sbus_req, error);
+            return;
+        }
+    }
+
+    list_ctx->dom = get_next_domain(list_ctx->dom, SSS_GND_DESCEND);
```

Comment:
The intention here is to gather result from all domains, right? I implemented `search_all_domains` plugin option for enumeration use case to allow `cache_req` to not stop on the first domain with result but get result from all domains.

This can be used in `ifp_users_list_by_name` instead of iteration, we just need to switch the filter plugins to this. 

We can also use it for the method implemented here. `cache_req` needs small work though to allow override plugin settings in similar way Fabiano did for PAM responder (`cache_req_data_set_bypass_cache`).